datanodes/src/main/scala/io/idml/IdmlObject.scala

```diff
@@ -41,5 +41,6 @@ abstract class IdmlObject extends IdmlValue {
   // scalastyle:on method.name
 
   override def toStringOption: Option[String] =
-    Some("{" + fields.flatMap { case (k, v) => v.toStringOption.map(k -> _) }.map { case (k, v) => s""""$k":$v""" }.mkString(",") + "}")
+    Some("{" + fields.toList.sortBy(_._1).flatMap { case (k, v) => v.toStringOption.map(k -> _) }.map { case (k, v) => s""""$k":$v""" }.mkString(",") + "}")
```

Comment:
and sort in the toString just in case (this is mostly used for groupBy on object keys)